### PR TITLE
Add Sandcastle.addToggleButton function

### LINKED ISSUE #4573: Shadows demo cleanup
The `Toggle` buttons in the Shadows demo should be checkboxes, same for the `Soft Shadows` button.  Right now it's not clear what is enabled/disabled at any given time.

### PR DESCRIPTION
**Pull request by fredj**
Fixes #4573

sandcastle_checkbox

### CODE REVIEW

**Comment by lilleyse** [comment]
Good start @fredj but typically we use knockout for these types of things. Check out the [Terrain demo](http://localhost:8080/Apps/Sandcastle/index.html?src=Terrain.html&label=Showcases), both the js and html tabs. You can also edit the css if it's hard to see the checkbox, like in [Atmosphere Color](http://localhost:8080/Apps/Sandcastle/index.html?src=Atmosphere%20Color.html&label=Showcases).

**Comment by hpinkos** [comment]
@lilleyse I think this is fine how it is since it's part of Sandcastle.  The code follows what we do for toolbar buttons and dropdowns

**Comment by lilleyse** [comment]
Ah yes you're right @hpinkos I read through it too fast. Someone more familiar with Sandcastle.js should review.

**Comment by fredj** [comment]
The CSS should be improved (vertical align, font size)

**Comment by hpinkos** [comment]
Great idea adding a Sandcastle helper function for this =)
Could you change the `Enable fog` checkbox in the terrain demo to use this as well?
The code looks good.  My only comment is that it'd be nice if it didn't expand all of the way to the width of the container.  I don't know if this is the best solution, but wrapping it in a `<button>` seems to fix that.

image
image

**Comment by mramato** [comment]
@emackey might have some better ideas here, I'm not sure wrapping it in a button makes sense.

**Comment by fredj** [comment]
code updated:
the label is wrapped in a button but I was able to remove the span (by using `label.appendChild(document.createTextNode(text));`)

**Comment by fredj** [comment]
> Could you change the Enable fog checkbox in the terrain demo to use this as well?

done, but there's many places in the demos where this function could be used (everywhere there's a "Toggle xxx" )

**Comment by hpinkos** [comment]
> but there's many places in the demos where this function could be used (everywhere there's a "Toggle xxx" )

Yes, please switch them over wherever you see fit.  The terrain demo was the only one I found when I quick scanned through, but it's great if you think it can be used other places.  Thanks!

**Comment by fredj** [comment]
I changed some demos but probably not all of them. I'm on vacation the next two weeks and I won't be able to continue to work on this.

The remaining one could be addressed in an other pull request.

File: CONTRIBUTORS.md [hunk 1]
```diff
@@ -65,6 +65,9 @@ See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how to contribute to Cesiu
    * [Andrey Orlov](https://github.com/AndreyOrlov)
 * [The Imagineers](https://www.theimagineers.com/)
    * [Heerco Grond](https://github.com/HeercoGrond)
+* [Camptocamp SA](https://www.camptocamp.com/)
+   * [Frédéric Junod](https://github.com/fredj)
+   * [Guillaume Beraudo](https://github.com/gberaudo)
```
Comment (fredj) [inline CONTRIBUTORS.md]: > it is OK for people to have their name in more than one spot

didn't know that, @gberaudo do you want to be moved or copied?

Comment (gberaudo) [inline CONTRIBUTORS.md]: @fredj, I am fine with copying.

Comment (fredj) [inline CONTRIBUTORS.md]: > @fredj, I am fine with copying.

ok, copied into the Camptocamp organization

**Comment by hpinkos** [comment]
@fredj this looks really great!  Thanks for updating all the other examples.
I just pushed a few tweaks to the button clicking.  Before the checkbox would only change if you clicked the checkbox or label, now it changes if you click anywhere on the button

**Pull request merged**